pkg/resource/apply.go

```diff
@@ -47,11 +52,99 @@ func (r *Helper) Apply(obj []byte) (ApplyResult, error) {
 		return "", err
 	}
 	defer r.deleteTempFile(fileName)
+
 	factory, err := r.getFactory("")
 	if err != nil {
 		r.logger.WithError(err).Error("failed to obtain factory for apply")
 		return "", err
 	}
+
+	// TODO: Replace with a cluster version check, if this cluster is 4.5+ and thus Kube 1.18+, we should have
+	// server side apply available, which is drastically faster than the local approach which currently involves a file
```

Comment:
It must not be the file write that is slow since the file write has already been done at this point, even when doing a server-side apply.

Reply:
I think I just botched this PR slightly, I am near certain I commented out the file write, but I did not carry that through to this commit somehow.